tests/test_server.py

```diff
@@ -851,6 +852,40 @@ def test_execute_with_not_existing_action(rasa_app: SanicTestClient):
     assert response.status == 500
 
 
+def test_trigger_intent(rasa_app: SanicTestClient):
+    data = {"name": "greet"}
+    _, response = rasa_app.post("/conversations/test_trigger/trigger_intent", json=data)
+
+    assert response.status == 200
+
+    parsed_content = response.json
+    assert parsed_content["tracker"]
+    assert parsed_content["messages"]
+
+
+def test_trigger_intent_with_missing_action_name(rasa_app: SanicTestClient):
+    test_sender = "test_trigger_intent_with_missing_action_name"
+
+    data = {"wrong-key": "greet"}
+    _, response = rasa_app.post(
+        f"/conversations/{test_sender}/trigger_intent", json=data
+    )
+
+    assert response.status == 400
+
+
+def test_trigger_intent_with_not_existing_intent(rasa_app: SanicTestClient):
+    test_sender = "test_trigger_intent_with_not_existing_intent"
+    _create_tracker_for_sender(rasa_app, test_sender)
+
+    data = {"name": "ka[pa[opi[opj[oj[oija"}
+    _, response = rasa_app.post(
+        f"/conversations/{test_sender}/trigger_intent", json=data
+    )
+
+    assert response.status == 500
```

Comment:
500? This would mean that the server runs into an error which is not handled? I'd prefer to handle it.